### PR TITLE
feat(core): update read_zkey with `new_unchecked`

### PR DESCRIPTION
**Pull request by vivianjeng**
According to https://github.com/zkmopro/mopro/issues/25#issuecomment-2067880755

| loading zkey time | ark-zkey | with `new_unchecked` |
|:--:|:--:|:--:|
| multiplier2 | 872.00ns |  1.21µs |
| keccak256 | 115.82s |  1.56s |

### CODE REVIEW

**Comment by oskarth** [comment]
I wonder if we want to combine ark-zkey with this? Because with ark-zkey we get 50% file size reduction due to point compression.

File: mopro-core/src/middleware/circom/mod.rs [hunk 1]
```diff
@@ -66,21 +67,21 @@ impl Default for CircomState {
...
 
-const ARKZKEY_BYTES: &[u8] = include_bytes!(env!("BUILD_RS_ARKZKEY_FILE"));
+// const ARKZKEY_BYTES: &[u8] = include_bytes!(env!("BUILD_RS_ARKZKEY_FILE"));
 
-// static ZKEY: Lazy<(ProvingKey<Bn254>, ConstraintMatrices<Fr>)> = Lazy::new(|| {
-//     let mut reader = Cursor::new(ZKEY_BYTES);
-//     read_zkey(&mut reader).expect("Failed to read zkey")
-// });
-
-static ARKZKEY: Lazy<(ProvingKey<Bn254>, ConstraintMatrices<Fr>)> = Lazy::new(|| {
```
Comment (oskarth) [inline mopro-core/src/middleware/circom/mod.rs:78]: do we have to remove the ARKZKEY stuff you think? or is this preferable?

thinking mostly about space savings which are quite significant

**Review by oskarth — APPROVED** [review]

**Comment by oskarth** [comment]
If we remove/don't update arkzkey stuff, could we create an issue explaining problem and current state? So even if we don't do it now we can introduce this for arkzkey too

**Comment by vivianjeng** [comment]
Create an issue here
https://github.com/zkmopro/mopro/issues/130

**Pull request merged**